pandas/_libs/reduction.pyx

```diff
@@ -59,16 +59,20 @@ cdef class _BaseGrouper:
             cached_typ = self.typ(
                 vslider.buf, dtype=vslider.buf.dtype, index=cached_ityp, name=self.name
             )
+            self.has_block = hasattr(cached_typ._mgr, "_block")
         else:
             # See the comment in indexes/base.py about _index_data.
             # We need this for EA-backed indexes that have a reference
             # to a 1-d ndarray like datetime / timedelta / period.
             object.__setattr__(cached_ityp, '_index_data', islider.buf)
             cached_ityp._engine.clear_mapping()
             cached_ityp._cache.clear()  # e.g. inferred_freq must go
-            object.__setattr__(cached_typ._mgr._block, 'values', vslider.buf)
-            object.__setattr__(cached_typ._mgr._block, 'mgr_locs',
-                               slice(len(vslider.buf)))
+            if self.has_block:
+                object.__setattr__(cached_typ._mgr._block, 'values', vslider.buf)
+                object.__setattr__(cached_typ._mgr._block, 'mgr_locs',
```

Comment:
This change is being done separately in https://github.com/pandas-dev/pandas/pull/40199